cypress/integration/ErrorHandling/MissingAction.spec.js

```diff
@@ -10,44 +10,66 @@ import * as editDialogModal from '../../support/components/EditDialogModal'
+  
+  context('Setup', () => {
+    it('Should import a model and wait for training to complete', () => {
+      models.ImportModel('z-missingAction', 'z-whatsYourName.cl')
+      modelPage.NavigateToTrainDialogs()
+      cy.WaitForTrainingStatusCompleted()
+    })
 
+    it('Should verify there are no error icons on the page and create a new Train Dialog', () => {
+      modelPage.VerifyNoErrorIconOnPage()
     train.CreateNewTrainDialog()
+    })
+  })
 
-    train.TypeYourMessage(common.gonnaDeleteAnAction)
-    editDialogModal.ClickScoreActionsButton()
-    train.SelectAction(common.whatsYourName)
-    
-    train.Save()
-
-    modelPage.NavigateToActions()
-    actions.DeleteAction(common.whatsYourName)
-    modelPage.NavigateToTrainDialogs()
-
-    modelPage.VerifyErrorIconForTrainDialogs()
-    train.VerifyErrorsFoundInTraining(`${common.validityIcon}${common.gonnaDeleteAnAction}`, common.gonnaDeleteAnAction, undefined)
-
-    train.EditTraining(`${common.validityIcon}${common.gonnaDeleteAnAction}`, common.gonnaDeleteAnAction, undefined)
+  context('Train', () => {
+    it('Should complete and save a simple 1 action Train Dialog', () => {
+      train.TypeYourMessage(common.gonnaDeleteAnAction)
+      editDialogModal.ClickScoreActionsButton()
+      train.SelectAction(common.whatsYourName)
+      train.Save()
+    })
+  })
 
-    editDialogModal.VerifyErrorMessage(trainDialogHasErrorsMessage)
+  context('Action', () => {
+    it('Should delete the action we just used in the Train Dialog', () => {
+      modelPage.NavigateToActions()
+      actions.DeleteAction(common.whatsYourName)
+    })
+  })
 
-    editDialogModal.SelectChatTurnStartsWith('ERROR: Can’t find Action Id')
-    editDialogModal.VerifyErrorMessage('Action does not exist')
-    scorerModal.VerifyMissingActionNotice()
+  context('Train Dialog Grid', () => {
+    it('Should verify there are now error icons showing in the Train Dialog grid', () => {
+      modelPage.NavigateToTrainDialogs()
+      modelPage.VerifyErrorIconForTrainDialogs()
+      train.VerifyErrorsFoundInTraining(common.gonnaDeleteAnAction, common.gonnaDeleteAnAction, '')
+    })
+  })
 
-    scorerModal.ClickAddActionButton()
-    actions.CreateNewAction({ response: common.whatsYourName, expectedEntities: 'name' })
+  context('Train', () => {
+    it('Should edit the Train Dialog and verify the expected error messages show up', () => {
+      train.EditTraining(common.gonnaDeleteAnAction, common.gonnaDeleteAnAction, '')
+      editDialogModal.VerifyErrorMessage(common.trainDialogHasErrorsMessage)
 
-    editDialogModal.VerifyNoErrorMessage()
+      editDialogModal.SelectChatTurnStartsWith('ERROR: Can’t find Action Id')
+      editDialogModal.VerifyErrorMessage('Action does not exist')
+      scorerModal.VerifyMissingActionNotice()
+    })
+  })
 
-    editDialogModal.ClickSaveCloseButton()
-    modelPage.VerifyNoErrorIconOnPage()
+  context('Train Add Action', () => {
```

Comment:
Looks like it's continue from next text so it's also Train Dialog text, Would put it with the one above.

Reply:
It is a little different, but I see your point so I moved it up and modified the test description slightly to better describe what that stage is doing.